Fix build with lablgtk 2.16.
Apply conditionally because it uses new types introduced in this version.

Index: coq-8.4/ide/preferences.ml
===================================================================
--- coq-8.4.orig/ide/preferences.ml
+++ coq-8.4/ide/preferences.ml
@@ -35,7 +35,7 @@ let mod_to_str (m:Gdk.Tags.modifier) =
     | `MOD5 -> "<Mod5>"
     | `CONTROL -> "<Control>"
     | `SHIFT -> "<Shift>"
-    |  `BUTTON1| `BUTTON2| `BUTTON3| `BUTTON4| `BUTTON5| `LOCK -> ""
+    |  `BUTTON1| `BUTTON2| `BUTTON3| `BUTTON4| `BUTTON5| `LOCK | `HYPER | `META | `RELEASE | `SUPER -> ""
 
 let mod_list_to_str l = List.fold_left (fun s m -> (mod_to_str m)^s) "" l
 
Index: coq-8.4/ide/utils/okey.ml
===================================================================
--- coq-8.4.orig/ide/utils/okey.ml
+++ coq-8.4/ide/utils/okey.ml
@@ -47,6 +47,10 @@ let int_of_modifier = function
   | `BUTTON3 -> 1024
   | `BUTTON4 -> 2048
   | `BUTTON5 -> 4096
+  | `HYPER -> 8192
+  | `META -> 16384
+  | `RELEASE -> 32768
+  | `SUPER -> 65536
 
 let print_modifier l =
   List.iter
